packages/webpack-cli/lib/bootstrap.js

```diff
@@ -71,6 +71,10 @@ async function runCLI(cliArgs) {
             parsedArgsOpts.entry = entry;
         }
 
+        if (parsedArgsOpts.bail && parsedArgsOpts.watch) {
+            logger.warn('You are using "bail" with "watch". "bail" will still exit webpack when the first error is found.');
+        }
```

Comment:
We should look at `compiler.options.bail` and `parsedArgsOpts.watch`, because other plugins can change options